pkg/types/ccipocr3/common_types.go

```diff
@@ -85,3 +86,29 @@ func (b *BigInt) UnmarshalJSON(p []byte) error {
 func (b BigInt) IsEmpty() bool {
 	return b.Int == nil
 }
+
+type P2PID [32]byte
+
+func (b P2PID) MarshalJSON() ([]byte, error) {
+	return json.Marshal(hex.EncodeToString(b[:]))
```

Comment:
P2P ID's should not be marshaled as hex, they have their own string encoding scheme please see libocr's PeerID type in ragep2p/types/types.go

Do we need to redefine the P2PID here? What's the win over using what we have in libocr?

Reply:
I remember trying the libocr's PeerID first and then changing my mind for some reason. Let me check again why.

Reply:
I was using `PeerID` from `github.com/smartcontractkit/chainlink/v2/core/services/p2p/types` and this seems like a cyclic dependency.

Reply:
Yeah don't import from core but you can import from libocr

Reply:
Bonus is that libocr already implements the required Marshal* methods so we don't need that boilerplate again